modin/pandas/base.py

```diff
@@ -122,6 +122,23 @@ class BasePandasDataset(ClassLogger):
     # but lives in "pandas" namespace.
     _pandas_class = pandas.core.generic.NDFrame
 
+    def _is_dataframe(self) -> bool:
+        """
+        Tell whether this is a dataframe.
+
+        Ideally, other methods of BasePandasDataset shouldn't care whether this
+        is a dataframe or a series, but sometimes we need to know. This method
+        is better than hasattr(self, "columns"), which for series will call
+        self.__getattr__("columns"), which requires materializing the index.
+
+        Returns
+        -------
+        bool : Whether this is a dataframe.
+        """
+        from .dataframe import DataFrame
+
+        return isinstance(self, DataFrame)
```

Comment:
```suggestion
        return issubtype(self._pandas_class, pandas.DataFrame)
```
we have a `._pandas_class` field, let's use that

Reply:
Done.